lib/view_component/translatable.rb

```diff
@@ -32,6 +32,25 @@ def build_i18n_backend
           )
         end
       end
+
+      def i18n_key(key, scope = nil)
+        scope = scope.join(".") if scope.is_a? Array
+        key = key&.to_s unless key.is_a?(String)
+        key = "#{scope}.#{key}" if scope
+        key = "#{i18n_scope}#{key}" if key.start_with?(".")
+        key
+      end
+
+      def translate(key = nil, **options)
```

Comment:
What do you think about having this method compile the component if it hasn't been compiled? That way we wouldn't have to expect folks to call `compile` first.